### DISCUSSION TITLE
Protractor to v7 Typescript conversion

### DISCUSSION THREAD

**Discussion by berdal84** [webdriverio/codemod#20] [Migration Issue]
Hello,

I just tried the last version (0.9.0) and I have some comments/suggestions to make.

- Is it possible to detect the appropriate parser (to avoid `--tsx`) since source extension is always `ts` ?
- Instead of displaying errors for `browser.actions()` (incompatible with wdio API), I suggest to comment those lines and maybe add a `// TODO: convert to wdio` (with the documentation link) and/or to add a `throw new Error('Convert the following commented lines to wdio.'); to avoid to miss it.
- Do you plan to convert types ? (ex: `ElementFinder` => `Promise<WebdriverIO.Element>`)
- I was importing ExpectedConditions as an alias EC, the transformation removed it and do not change it's usage: `browser.waitUntil(EC.presenceOf(toolBtn))` Do you plan to handle this later ?

Thanks in advance

**Comment by berdal84** [comment]
For my last point, I just saw the alias is not the problem. Translation is manual I assume.

**Comment by christian-bromann** [comment]
> * Is it possible to detect the appropriate parser (to avoid `--tsx`) since source extension is always `ts` ?

Good idea, added as requirement for #11 

> * Instead of displaying errors for `browser.actions()` (incompatible with wdio API), I suggest to comment those lines and maybe add a `// TODO: convert to wdio` (with the documentation link) and/or to add a `throw new Error('Convert the following commented lines to wdio.'); to avoid to miss it.

Yeah, so rather than let the compiler fail, have it add a throw exception. This way it makes it easier to test for sure. Mind creating an issue for it?

> * Do you plan to convert types ? (ex: `ElementFinder` => `Promise<WebdriverIO.Element>`)

Let's create an issue for it to add TypeScript types where ever possible.

> * I was importing ExpectedConditions as an alias EC, the transformation removed it and do not change it's usage: `browser.waitUntil(EC.presenceOf(toolBtn))` Do you plan to handle this later ?

So currently the mod translates:

```js
const EC = protractor.ExpectedConditions;
```
into
```js
const EC = require('wdio-wait-for')
```

which is our drop in replacement. Are you saying you import it this way:

```js
import { EC } from 'protractor'
```

?

**Reply by berdal84** [reply]
> 
> which is our drop in replacement. Are you saying you import it this way:
> 
> ```js
> import { EC } from 'protractor'
> ```
> 
> ?

I was importing like that:

```ts
import { ExpectedConditions as EC } from 'protractor';
```

**Reply by berdal84** [reply]
> > * Instead of displaying errors for `browser.actions()` (incompatible with wdio API), I suggest to comment those lines and maybe add a `// TODO: convert to wdio` (with the documentation link) and/or to add a `throw new Error('Convert the following commented lines to wdio.'); to avoid to miss it.
> 
> Yeah, so rather than let the compiler fail, have it add a throw exception. This way it makes it easier to test for sure. Mind creating an issue for it?

Sure

**Reply by christian-bromann** [reply]
> I was importing like that:

Let's make an issue for that as well
